lsp-methods.el

```diff
@@ -545,9 +545,13 @@ If NO-WAIT is non-nil, don't synchronously wait for a response."
   "Send BODY as a request to the language server, and call CALLBACK with
 the response recevied from the server asynchronously."
   (let ((client (lsp--workspace-client lsp--cur-workspace))
+        (buf (current-buffer))
         (id (plist-get body :id)))
     (cl-assert id nil "body missing id field")
-    (puthash id callback (lsp--client-response-handlers client))
+    (puthash id (lambda (result)
+                  (when (buffer-live-p buf)
```

Comment:
clients might still be interested in the result even if the buffer is closed. E. g. if we do "did save handling" async the original buffer would be deleted.

Reply:
I propose explicit optional parameter(s) which turn on/turn off this feature.